quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakReadyHealthCheckTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package test.org.keycloak.quarkus.services.health;
+
+import io.quarkus.test.QuarkusUnitTest;
+import org.hamcrest.Matchers;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.sql.SQLException;
+
+import static io.restassured.RestAssured.given;
+
+public class KeycloakReadyHealthCheckTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest test = new QuarkusUnitTest()
```

Comment:
I believe one more test - simulate that a database is down and check if the Readiness Probe returns a failure.

I believe the easiest way to do it is to modify the way we wire the Agroal bean. Maybe we should create an adapter there (e.g. `DelegatingDatasource`) and inject a mock during testing there.

Reply:
You should be able to inject the datasource bean and then try to call close (or reset metrics). I did a quick test here and it seems to work.

You may want a separate test for down status though.

Reply:
Clever idea @pedroigor ! Thanks for the hint.

Together with @miquelsi we agreed to see if we could somehow inject a mock there to manipulate the metrics. This would allow us to test all crazy scenarios in the future. If that won't work - we agreed to try killing the H2 DB (essentially calling `#close()` as you suggested). We'll see how it goes.

Reply:
@slaskawi It may be an option to not boot the application at all (considering we have integration already) for more advanced scenarios so you can just set some impl directly to the health check class. It should help with crazy scenarios and much easier to implement.

Adding a bean to just allow testing seems too much because at runtime we only need the real datasource. You may also consider `@Priority` annotation. Not sure if it may help here.

Reply:
Thanks for the thip @pedroigor. I've added a new test class for negative cases.